src/server/qgsfcgiserverresponse.cpp

```diff
@@ -54,39 +58,52 @@ typedef struct QgsFCGXStreamData
 } QgsFCGXStreamData;
 #endif
 
-QgsSocketMonitoringThread::QgsSocketMonitoringThread( bool *isResponseFinished, QgsFeedback *feedback )
-  : mIsResponseFinished( isResponseFinished )
-  , mFeedback( feedback )
+// to be able to use 333ms expression as a duration
+using namespace std::chrono_literals;
+
+// used to synchronize socket monitoring thread and fcgi response
+std::timed_mutex gSocketMonitoringMutex;
```

Comment:
Can we have it member of MonitoringThread object instead and unlock it when setResponseFinished is called with true before updating mIsResponseFinished ?

Or even rename setResponseFinished(bool) with something like stop ?

Reply:
done!